### PR TITLE
check validity of presets on startup

### PR DESCRIPTION
**Pull request by cymed**
On startup, we should check if the database is suitable for production

### CODE REVIEW

File: plugin/teksi_wastewater/utils/tww_validity_check.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    cursor.execute("SELECT prefix FROM tww_sys.oid_prefixes WHERE active;")
+    prefixes = cursor.fetchall()
+    active_pref = prefixes[0].prefix
+    msgtxt = []
+    if len(prefixes) > 1:
+        msgtxt.append(
+            "more than one oid_prefix set to active. Generation of Object-ID will not work"
+        )
+    if active_pref == "ch000000":
+        msgtxt.append("OID prefix set to 'ch000000'. Database not safe for production")
```
Comment (sjib) [inline plugin/teksi_wastewater/utils/tww_validity_check.py:32]: Maybe add: Set the OID prefix for your production database to active!

**Comment by sjib** [comment]
@cymed Wow - nice new feature!

File: plugin/teksi_wastewater/utils/tww_validity_check.py (hunk 1 above)
Comment (cymed) [reply plugin/teksi_wastewater/utils/tww_validity_check.py:32]: solved in f0251af

**Comment by sjib** [comment]
For me this looks good. May be the (TEKSI) service providers want to comment also.
@varrieta @ponceta @tproduit @Fabien-Cattin @gsollberger @triformmoritz

**Review by sjib — APPROVED** [review]

**Review by ponceta — APPROVED** [review]
Very good input indeed.

File: plugin/teksi_wastewater/teksi_wastewater_plugin.py [hunk 2]
```diff
@@ -310,6 +311,25 @@ def initGui(self):
...
+        if not pg_layer:
+            self.iface.messageBar().pushMessage(
+                "Error",
+                "Could not determine the Postgres connection information. Make sure the TWW project is loaded.",
+                level=Qgis.Critical,
+            )
+
+        pgservice = pg_layer.dataProvider().uri().service()
+        msgs = tww_check_oid_prefix(pgservice)
+        msgs.extend(tww_check_oid_prefix(pgservice))
```
Comment (domi4484) [inline plugin/teksi_wastewater/teksi_wastewater_plugin.py:325]: @cymed should that be `msgs.extend(tww_check_fk_defaults(pgservice)`?

**Pull request merged**